pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go

```diff
@@ -91,6 +92,14 @@ func Add(mgr manager.Manager) error {
 	}
 	clusterInfo.SCCAvailable = sccAvailable
 
+	addMonitorServiceResources, err := IsMonitoringAvailable(clientset)
+	if err != nil {
+		// we don't want CNAO to fail only if monitoring cannot be activated.
+		addMonitorServiceResources = false
+		fmt.Printf("failed to check for availability of Monitoring namespace: %v", err)
```

Comment:
it could be nice to have, but since statusManager (and correspondingly the eventEmitter) are only initialized when the reconciler start to work and this code runs before that, then I'm not sure it's worth the trouble.